### PR TITLE
feat(A529yyds):add pgsql operation functions:register, login and chan…

### PR DESCRIPTION
**Pull request by A529yyds**
…gerpassword

fix(A529yyds):fixed pgsqlClient __conn initialization

### CODE REVIEW

File: src/Order.cpp [hunk 1]
```diff
@@ -110,12 +139,144 @@ nlohmann::json Redis::SetCheckCode::constructResponse(const nlohmann::json &orde
...
+    if(sqlExec(cmd, memoryData, reply))
+    {
+        result["success"] = 1;
+        std::string searchPwd = reply.size() > 0 ? reply[0][2].as<std::string>() : "";
+        if(searchPwd == order.find("password").value().get<std::string>())
+        {
+            result["correct"] = 1;
+        }
+    }
+    LOG(ERROR) << result.dump();
```
Comment (yushou-cell) [inline src/Order.cpp]: There is a problem here, both success and failure will print error logs

File: src/Order.cpp [hunk 2]
```diff
@@ -110,12 +139,144 @@ nlohmann::json Redis::SetCheckCode::constructResponse(const nlohmann::json &orde
...
+bool SQL::SQLOperation::sqlExec(std::string cmd, pgsqlClient &memoryData, pqxx::result &reply)
+{
+    bool ret = false;
+   try{
+        reply = memoryData.execCommandOneSql(cmd);
+        ret = true;
+    }
+    catch(const pqxx::sql_error &e)
+    {
+        LOG(ERROR) << "SQL error: " << e.what() << std::endl;
```
Comment (yushou-cell) [inline src/Order.cpp:159]: In this exception, a JSON command-reply needs to be constructed indicating execution failure, with success=0 and correct=0. The same applies below.

File: src/Order.cpp [hunk 3]
```diff
@@ -110,12 +139,144 @@ nlohmann::json Redis::SetCheckCode::constructResponse(const nlohmann::json &orde
...
+{
+}
+
 SQL::Register::Register()
 {
 }
 
+bool SQL::SQLOperation::isOrderRight(const nlohmann::json &order, std::string orType)
+{
+    return order.find("order").value().get<std::string>() == orType;
```
Comment (yushou-cell) [inline src/Order.cpp]: The function ConstructResponseMsgPgSQL has been validated. There’s no need for validation here. This function can be removed

Comment (A529yyds) [reply src/Order.cpp]: OK

**Review by yushou-cell — APPROVED** [review]
Approve

**Pull request merged**